store.go

```diff
@@ -2836,6 +2836,13 @@ func (s *store) mount(id string, options drivers.MountOpts) (string, error) {
 	if rlstore.Exists(id) {
 		return rlstore.Mount(id, options)
 	}
+	// check if the layer is in a read-only store, and return a better error message
+	for _, store := range lstores {
+		if store.Exists(id) {
+			return "", fmt.Errorf("layer present in a read-only store: %w", ErrLayerUnknown)
```

Comment:
If I saw this error, I would ask, but I don't want to write to the layer only examine it.  Could we return the layer but marked as read/only?

Reply:
with MountImage we always mount in read-only mode, but to mount it we still need to modify the store and need to grab a writeable lock there.

I don't see an immediate way to "copy up" the layer in a transparent way, we would need the equivalent of `CreateContainer` and reference the read-only layer as a lower layer but we will end up with two different layers.  I think it is cleaner to handle it manually with `CreateContainer` at least it is clear what is happening in the store

Reply:
How about "mounting read/only store images is not allowed"

Reply:
sure, fixed and pushed a new version